typescript/sdk/src/providers/SmartProvider/SmartProvider.ts

```diff
@@ -310,10 +341,21 @@ export class HyperlaneSmartProvider
               ...providerMetadata,
             },
             `Error from provider.`,
-            isLastProvider ? '' : 'Triggering next provider.',
           );
           providerResultErrors.push(result.error);
-          pIndex += 1;
+
+          const nonRetryable = hasNonRetryableError({
+            method,
+            error: result.error as Error,
+          });
+
+          pIndex += nonRetryable
+            ? providers.length // Setting this to providers.length will force exit the loop and not retry with the next provider
+            : 1;
+
+          this.logger.debug(
+            isLastProvider || nonRetryable ? '' : 'Triggering next provider.',
```

Comment:
Should we log something about it being the last time it's attempted and why? feels useful to know if it's non retryable